core-common/src/main/java/org/glassfish/jersey/message/internal/FormMultivaluedMapProvider.java

```diff
@@ -53,9 +54,8 @@ public FormMultivaluedMapProvider() {
 
     @Override
     public boolean isReadable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
-        // Only allow types MultivaluedMap<String, String> and MultivaluedMap.
-        return type == MultivaluedMap.class
-                && (type == genericType || mapType.equals(genericType));
+        return (type == MultivaluedMap.class && (type == genericType || mapType.equals(genericType))) 
+                || (type == Map.class || genericType == Map.class);
```

Comment:
I think the comment was nice to have - should be extended for Map.
I am not quite sure about the `genericType == Map.class` part. 
As `MultivaluedMap` is defined as `MultivaluedMap<K, V> extends Map<K, List<V>>` the expected generic type for this case would be `Map<String,List<String>>`. Hard to get a proper generic `Type` of that without declaring a type that extends this and get it from there. Maybe it is easier to drop the check or to do it by decomposing the `genericType` passed.